Updated code tutorial (#77)
ZebraDevs/Zebra-Iota-Edge-SDK: docs/Tutorial.md

# IOTA Identity Tutorial

This tutorial demonstrates the use of the IOTA Identity Framework to create, share and verify some information about a device in the following sections:

1. Initialize an IOTA Identity client
2. Create identities
3. Create verifiable credentials
4. Create verifiable presentations
5. Verify presentations
6. Putting it all together

For each step, a function is defined that you should be able to reuse in your application(s).

## Before you begin

> If you are unfamiliar with W3C decentralized identifiers and verifiable credentials, we recommend reading the overviews of chapters 2, 3.1 and 3.2 of the [IOTA Identity guide](https://wiki.iota.org/identity.rs/introduction).

Reviewer: ```suggestion
If you are unfamiliar with W3C Decentralized Identifiers (DID) and Verifiable Credentials (VCs), we recommend reading the overviews of chapters 2, 3.1 and 3.2 of the [IOTA Identity guide](https://wiki.iota.org/identity.rs/introduction).
```

The IOTA Identity framework is implemented in Rust. In order to use it in JavaScript, WASM bindings are available from the NPM package `@iota/identity-wasm`. Please refer to [IOTA Identity WASM - Getting Started](https://wiki.iota.org/identity.rs/libraries/wasm/getting_started) for set up instructions in both NodeJS and web (browser) contexts. This tutorial assumes a web context, but code comments show where usage is not the same as in NodeJS.

To complete the tutorial, you will need the following imports:

```js
import {
    init,
    Client,
    Config,
    Digest,
    Document,
    KeyCollection,
    KeyPair,
    KeyType,
    Network,
    VerifiableCredential,
    VerifiablePresentation,
    VerificationMethod
} from "@iota/identity-wasm/web"; // NOTE: Web context.
```

## Initialize an IOTA Identity Client

> We will create a client that connects to the IOTA mainnet through one of the IOTA Foundation's nodes. Data transactions (such as identities) are pruned from the ledger state after some time and so we should also configure a permanode from which identities can be fetched by the client even after a pruning cycle.

Reviewer: ```suggestion
We will create a client that connects to the IOTA mainnet through one of the IOTA Foundation's nodes. Data transactions (such as identities) are pruned after some time and so we should also configure a Permanode from which identities can be fetched by the client even after a pruning cycle.
```

```js
async function createClient() {
    // Configure IOTA Identity client to connect to the IOTA mainnet.
    const cfg = Config.fromNetwork(Network.try_from_name("main"));
    cfg.setNode("https://chrysalis-nodes.iota.org");
    cfg.setPermanode("https://chrysalis-chronicle.iota.org/api/mainnet/");
    
    // Return the client.
    return Client.fromConfig(cfg);
}
```

## Create Identities

> Creating an identity involves generating a public-private key pair (`authKeyPair`) that gets used to control access to the DID document on the Tangle. Additionally, a separate key collection is created and added as a verification method for this identity so verifiers may authenticate the DIDs contained in VCs and VPs. The `client` parameter can be provided using `createClient` above.

Reviewer: ```suggestion
Creating an identity involves generating a public-private key pair (`authKeyPair`) that gets used to control access to the DID document on the Tangle. Additionally, a separate key collection is created and added as a verification method for this identity so verifiers may authenticate VCs issued. The `client` parameter can be provided using `createClient` above.
```
